static/flow/views/program-editor-panel.js

```diff
@@ -838,7 +836,7 @@ var ProgramEditorPanel = function(options) {
         if (block) {
             _this.undisplayBlock(block);
             _this.m_diagram.removeBlock(block);
-            delete _this.nameHash[block.name];
```

Comment:
@dstrawberrygirl There were a variety of places where we were trying to keep the nameHash object up to date, but a detailed review showed that there were cases where our code was not complete and the nameHash object was not up to date.  I added an updateNameHash() function which should be used to clean and rebuild the nameHash object (it's a fast operation).